### PR TITLE
feat: Svg network mobile

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**

hardcode a mapping of chainIDs to currency symbol, once the svg are uploaded we mapped the images to the network name and chainIDs

Acceptance criteria
When the network list is displayed within the Select a network modal, or custom network list, or any similar context, the hardcoded network image avatars should be displayed. This will reduce the often appearing "?" that appears when the network avatar is not present.

## **Related issues**

Fixes: [#1813](https://github.com/MetaMask/MetaMask-planning/issues/1813)

## **Manual testing steps**

Go to the network modal
Add a new network displayed on this [list](https://docs.google.com/spreadsheets/d/16G3SRiQvWXK6hp5duob0aopeKyx2HOAQpz152Fy8VNQ/edit#gid=2080069111)
check if the image is displayed

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/26223211/b50efcc2-37dc-43a1-b506-ab6f72e1b00e

### **After**

https://github.com/MetaMask/metamask-mobile/assets/26223211/9f29ceea-45cb-409e-93af-0824d9ea7373

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/Ramp/Views/NetworkSwitcher/NetworkSwitcher.tsx [hunk 1]
```diff
@@ -264,6 +272,9 @@ function NetworkSwitcher() {
                           size={AvatarSize.Sm}
                           name={'Linea Mainnet'}
                           imageSource={imageIcons['LINEA-MAINNET']}
+                          chainId={convertHexToDecimal(
+                            selectedChainId,
+                          ).toString()}
```
Comment (wachunei) [inline app/components/UI/Ramp/Views/NetworkSwitcher/NetworkSwitcher.tsx:277]: This is not needed, `selectedChainId` is already a base 10 string.

File: app/components/UI/Ramp/Views/NetworkSwitcher/NetworkSwitcher.tsx [hunk 2]
```diff
@@ -239,6 +244,9 @@ function NetworkSwitcher() {
                           size={AvatarSize.Sm}
                           name={'Ethereum Mainnet'}
                           imageSource={imageIcons.ETHEREUM}
+                          chainId={convertHexToDecimal(
+                            selectedChainId,
+                          ).toString()}
```
Comment (wachunei) [inline app/components/UI/Ramp/Views/NetworkSwitcher/NetworkSwitcher.tsx:249]: This is not needed, `selectedChainId` is already a base 10 string.

**Review by wachunei — CHANGES_REQUESTED** [review]

File: app/component-library/components/Badges/Badge/Badge.test.tsx [hunk 3]
```diff
@@ -13,11 +13,16 @@ import {
...
 
 describe('Badge', () => {
   it('should render badge network given the badge network variant', () => {
     const wrapper = shallow(
-      <Badge {...SAMPLE_BADGENETWORK_PROPS} variant={BadgeVariant.Network} />,
+      <Badge
+        {...SAMPLE_BADGENETWORK_PROPS}
+        variant={BadgeVariant.Network}
+        chainId={NetworkId.mainnet}
+      />,
```
Comment (wachunei) [inline app/component-library/components/Badges/Badge/Badge.test.tsx:25]: The chainId prop was not part of the test before and it was already defined in `SAMPLE_BADGENETWORK_PROPS`, so probably the chain needed is only in [`BadgeNetwork.constants`](https://github.com/MetaMask/metamask-mobile/pull/8722/files#diff-ff563c4e4aa2e562fa31a3c72c505d881c659a04dabfccaa5a2c33731df3b5b8).

**Comment by brianacnguyen** [comment]
also please break this PR up. It's HUGE

File: app/component-library/components/Networks/Network/Network-icon.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { ViewProps } from 'react-native';
+import { NETWORKS_CHAIN_ID_WITH_SVG } from '../../../../constants/network';
+
+import { networkByChainID } from './Network.assets';
+
+/**
+ * Network icon component props.
+ */
+export interface NetworkIconProps extends ViewProps {
```
Comment (brianacnguyen) [inline app/component-library/components/Networks/Network/Network-icon.tsx:10]: Put in .type file

File: app/component-library/components/Networks/Network/Network-icon.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  name?: string;
```
Comment (brianacnguyen) [inline app/component-library/components/Networks/Network/Network-icon.tsx:11]: Comment with description for name

File: app/component-library/components/Networks/Network/Network-icon.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  onError?: (arg: boolean) => void;
```
Comment (brianacnguyen) [inline app/component-library/components/Networks/Network/Network-icon.tsx:12]: Comment with description for onError

File: app/component-library/components/Networks/Network/Network-icon.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  /**
+   * Enum to select between networks chainId.
```
Comment (brianacnguyen) [inline app/component-library/components/Networks/Network/Network-icon.tsx:14]: Optional. Also try to keep the descriptions consistent

File: app/component-library/components/Networks/Network/Network-icon.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+}: NetworkIconProps) => {
+  if (!chainId) {
+    return null;
+  }
+
+  const Svg = networkByChainID[chainId];
+  return <Svg name={name || ''} {...props} width="100%" height="100%" />;
+};
+
+export default NetworkIcon;
```
Comment (brianacnguyen) [inline app/component-library/components/Networks/Network/Network-icon.tsx:34]: Add test and stories

File: app/component-library/components/Networks/Network/Network.assets.ts [hunk 9]
```diff
@@ -0,0 +1,117 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (brianacnguyen) [inline app/component-library/components/Networks/Network/Network.assets.ts:1]: How is this library going to be maintained?

File: app/component-library/components/Networks/Network/Network.types.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { NETWORKS_CHAIN_ID_WITH_SVG } from '../../../../constants/network';
```
Comment (brianacnguyen) [inline app/component-library/components/Networks/Network/Network.types.ts:1]: Please group imports accordingly

File: app/component-library/components/Pickers/PickerNetwork/PickerNetwork.types.ts [hunk 11]
```diff
@@ -13,6 +13,10 @@ export interface PickerNetworkProps extends TouchableOpacityProps {
    * Network label to display.
    */
   label: string;
+  /**
+   * Network chainId.
```
Comment (brianacnguyen) [inline app/component-library/components/Pickers/PickerNetwork/PickerNetwork.types.ts:17]: Keep this consistent

File: app/component-library/components/Pickers/PickerNetwork/PickerNetwork.tsx [hunk 12]
```diff
@@ -33,6 +35,7 @@ const PickerNetwork = ({
         size={AvatarSize.Xs}
         name={label}
         imageSource={imageSource}
+        chainId={convertHexToDecimal(chainId).toString()}
```
Comment (brianacnguyen) [inline app/component-library/components/Pickers/PickerNetwork/PickerNetwork.tsx:38]: Why is this not part of the original component?

File: app/component-library/components/Toast/Toast.tsx [hunk 13]
```diff
@@ -156,6 +160,7 @@ const Toast = forwardRef((_, ref: React.ForwardedRef<ToastRef>) => {
             imageSource={networkImageSource}
             size={AvatarSize.Md}
             style={styles.avatar}
+            chainId={convertHexToDecimal(chainId).toString()}
```
Comment (brianacnguyen) [inline app/component-library/components/Toast/Toast.tsx:163]: same

File: app/images/acala-logo.svg [hunk 14]
```diff
@@ -0,0 +1,13 @@
+<svg width="100%" height="100%" viewBox="0 0 128 128" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (brianacnguyen) [inline app/images/acala-logo.svg:1]: Please optimize SVGs

**Review by brianacnguyen — CHANGES_REQUESTED** [review]

**Pull request closed without merging**